### ISSUE TITLE
Flux function last() not working as expected with InfluxDB 2

### ISSUE THREAD

**Issue by schamock** [influxdata/influxdb#20486]
__Steps to reproduce:__
1. Use the Flux function `last()`in a minimal statement

__Expected behavior:__
I tried to get the latest value of a measurement:
```
from(bucket: "abucket")
  |> range(start: 2021-01-10T23:34:30Z, stop:2021-01-11T10:34:30Z)
  |> filter(fn: (r) => r._measurement == "measurement")
  |> filter(fn: (r) => r._field == "some value")
  |> last()
```
In this measurement I have values every second (confirmed with the same statement but without `last()`.
I expected to get a value of 10:34:30 on 2021-01-11

__Actual behavior:__
I'm getting the value from 23:59:59 on 2021-01-10. However if I insert a `|> keep(columns: ["_time", "_value"])`before the `|> last()`then everything is working as expected (this command will remove _start, _stop, _measurement and _field in this example).

__Environment info:__

* System info: Linux 5.10.5-v8+ aarch64
* InfluxDB version: Influx CLI 2.0.3 (git: fe04d346df) build_date: 2020-12-15T01:00:16Z
* Other relevant environment details: probably nothing relevant

__Config:__
Standard setup after using the upgrade script from 1.8 to 2.0

**Comment by schamock** [comment]
After some more testing I found out, that this behavior doesn't show up with all start/stop combinations. There seams to something with 23:59:59 of the 10th of January... When setting the start date to somewhere on the 11th then everything is fine?!
This behavior with this specific start/stop combination is reproducible in my config with different buckets and measurements

**Comment by shaoyucheng** [comment]
I met the same problem in my situation, and i also found that in this case, you would get the same result whether used last() or first() which was not reasonable. finally i added group() before last() and got the expected output. FYI

**Comment by R-Studio** [comment]
**Same issue here with InfluxDB 2.0.2 and also with fresh installed InfluxDB 2.0.3**!

**Data-Description**
- We have two hosts as influx-tags (host = "FPPW9607" & host = "FPPW9603")
- We have to fields ("Read_Bytes_persec" & "Write_Bytes_persec")
- The VM-Disk ("C:-ClusterStorage-Silber-CSV02_C1TN9601-XPPW0156-INT_WINDOWS_SERVER_2019_DX.VHDX") was moved from FPPW9607 to FPPW9603

**What I want / What I expect**
I want the last "_value" for the fields "Read_Bytes_persec" & "Write_Bytes_persec" foreach "instance"
This should be the following (field, timestamp, value):
- Read_Bytes_persec, **2021-01-30 23:58:31 GMT+1, 0**
- Write_Bytes_persec, **2021-01-30 23:58:31 GMT+1, 20570.056640625**

**What I get / Actual behavior**
I get data from the wrong timestamp:
- Read_Bytes_persec, **2021-01-30 15:01:31 GMT+1**, 0
- Write_Bytes_persec, **2021-01-30 15:01:31 GMT+1**, 0
-> Why data from _2021-01-30 15:01:31 GMT+1_ instead of data from _2021-01-30 23:58:31 GMT+1_?

**Steps to reproduce**
- Download [demodaten.txt](https://github.com/influxdata/influxdb/files/5904800/demodaten.txt)
- Rename the file to _demodaten.csv_ 
- Import the csv in InfluxDB 2.x: `influx write -b <YOUR_BUCKET> -f demodaten.csv -o <YOUR_ORG> -t <YOUR_TOKEN>`
- Run the following Flux Query:
```
from(bucket: "<YOUR_BUCKET>")
  |> range(start: -30d, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["_measurement"] == "hyperv_storage")
  |> group(columns: ["instance", "_field"])
  |> last()
```

**Something is really strange here!**
I could imagine **this issue has a relation to the length of an tag-value**, in this example for the tag "instance". Because if I rename it for example from 
_C:-ClusterStorage-Silber-CSV02_C1TN9601-XPPW0156-INT_WINDOWS_SERVER_2019_DX.VHDX_ 
to 
_C:-ClusterStorage-Silber-CSV02_C7TN9991-VM-DISK2019_DX.VHDX_ 
the issue doesn't occurs.

**Comment by R-Studio** [comment]
Maybe this issue is also related to:
- https://community.influxdata.com/t/recent-edition-of-influxdb-2-0-2-last-function-not-work-as-expected/16942
- https://github.com/influxdata/influxdb/issues/20595

**Comment by R-Studio** [comment]
@schamock Unfortunately also with `|> keep(columns: ["_time", "_value"])` it does not work for me. :(
Do you found another workaround?

**Comment by schamock** [comment]
Hi @R-Studio,
sorry for replying late... I just found out, that this "keep" worked for me. I can't even tell, if this is robust or fails under certain circumstances. Unfortunately I have the same problem with version 2.0.4 :(

**Comment by NektoDron** [comment]
last(column: "_time") works right for me
default last() returns a garbage...

**Comment by R-Studio** [comment]
@danxmoran any news on this?

**Comment by danxmoran** [comment]
Possible duplicate of influxdata/flux#3480

**Comment by danxmoran** [comment]
@R-Studio it's actively being worked on by a member of the Flux team, so we should hopefully have a fix soon

**Comment by R-Studio** [comment]
@danxmoran thanks for your reply. 

Can this issue maybe also affects the function "**stateChangeOnly()**" https://github.com/influxdata/flux/issues/3519? 
_(My thoughts: the function uses internally the function "last()")_

**Comment by danxmoran** [comment]
Fixed by #21140
